### ISSUE TITLE
10$: Check variable references in enum initializers

### ISSUE THREAD

**Issue by strager** [quick-lint/quick-lint-js#756]
Currently, quick-lint-js reports no diagnostics in the following code, but it should:
```typescript
enum E {
  A = B,  // error: use of enum member 'B' before declaration
  B,
  C = D,  // warning: use of undeclared variable 'D'
}
enum E {
  D,
  F = B,
  G = undeclaredVar, // warning: use of undeclared variable 'undeclaredVar'
}
```
